R/direct_model.R

```diff
@@ -35,23 +24,14 @@ as_direct_model.data.frame <- function(data) {
   class(data) <- c("epidist_direct_model", class(data))
   data <- data |>
     mutate(delay = .data$stime - .data$ptime)
-  epidist_validate(data)
+  epidist_validate_model(data)
   return(data)
 }
 
-#' Validate direct model data
-#'
-#' This function checks whether the provided `data` object is suitable for
-#' running the direct model. As well as making sure that
-#' `is_direct_model()` is true, it also checks that `data` is a `data.frame`
-#' with the correct columns.
-#'
-#' @param data A `data.frame` containing line list data
-#' @param ... ...
-#' @method epidist_validate epidist_direct_model
+#' @method epidist_validate_model epidist_direct_model
 #' @family direct_model
 #' @export
-epidist_validate.epidist_direct_model <- function(data, ...) {
+epidist_validate_model.epidist_direct_model <- function(data, ...) {
```

Comment:
you can just stick to having one validate function as the class of input can be used i.e direct_model vs linelist